src/rgw/rgw_cache.cc

```diff
@@ -322,3 +337,423 @@ void ObjectCache::chain_cache(RGWChainedCache *cache) {
   chained_cache.push_back(cache);
 }
 
+int cacheAioWriteRequest::create_io(bufferlist& bl, unsigned int len, string oid) {
```

Comment:
This also seems poorly factored. That the actual details of fetching information from the disc cache is moved into RGW while the interfaces are part of librados. It would be better if either the whole thing were a caching library on top of librados that RGW could use or if it were specifically a part of RGW and didn't modify librados.

From this commit, the latter seems the best approach.